pylxd/models/storage_pool.py

```diff
@@ -698,6 +698,13 @@ def get(cls, volume, name):
         if response.json()["metadata"]["expires_at"] == "0001-01-01T00:00:00Z":
             snapshot.expires_at = None
 
+        # This field is normally empty so derive it from its volume.
+        snapshot.content_type = volume.content_type
+
+        # When dealing with older API version the response parsing can be messy, so we set those values to None.
+        if not isinstance(snapshot.created_at, str):
+            snapshot.created_at = None
```

Comment:
There is an API extension named `storage_volumes_created_at`, which might be of interest here. It's not available on 4.0 nor 5.0 though. But maybe you can still make use of it?

Reply:
I'd prefer checking the response object because:
 - 5.0 does not include the extension but does include `created_at` on the response;
 - pyLXD does not refresh host info that often so checking for API extensions is not resilient to updates.

Reply:
> I'd prefer checking the response object because:
> 
>     * 5.0 does not include the extension but does include `created_at` on the response;

Looks like a partial backport then. @tomponline as the original author of the feature and now the release manager, do you think we should leave things as is on LXD `stable-5.0`?

>     * pyLXD does not refresh host info that often so checking for API extensions is not resilient to updates omLXD.

pyLXD should check extension support at connection time so LXD won't change during the lifetime of that connection.

Reply:
> pyLXD should check extension support at connection time so LXD won't change during the lifetime of that connection.

Sorry for the confusion, I did mean this as an update during the lifetime of a connection. I realise this is not a problem but thought I should mention as a bonus advantage of this approach.

Reply:
> > I'd prefer checking the response object because:
> > ```
> > * 5.0 does not include the extension but does include `created_at` on the response;
> > ```
> 
> Looks like a partial backport then. @tomponline as the original author of the feature and now the release manager, do you think we should leave things as is on LXD `stable-5.0`?
> 
> > ```
> > * pyLXD does not refresh host info that often so checking for API extensions is not resilient to updates omLXD.
> > ```
> 
> pyLXD should check extension support at connection time so LXD won't change during the lifetime of that connection.

We cant add db schema changes which is why the field isnt in the db. It is in the api structs as we tend to keep them in sync as much as possible (this is what stephane told me).

But iirc the field shouldn't be back ported.

If there is an api extension then its right to check for it.

Reply:
Indeed the field is included on the response but it is always filled with the default value. I will then check for the extension for correctness.

Reply:
OK so with LXD 5.0 the extension field is not there but the `created_at` is there but always null. Make sense, thank you both for clarifying.